_drafts/commands.md

```diff
@@ -0,0 +1,74 @@
+---
+state: In Progress
+---
+
+# Commands
+
+"Commands" are an explicit interface for defining discrete pieces of functionality in a Hubot script. Initially, a command is a subset of the existing `robot.respond` interface. Over time, commands will start to gain more power and will be exposed through a command line interface or as native "slash commands" on services that support it, like Slack.
+
+## Motivation
+
+Most Hubot scripts have a command-ish interface, but the semantics of that interface are left to the whim of the script author. This can lead to confusing or conflicting scripts with unexpected behavior.
+
+Here are some of the problems with `robot.hear` and `robot.respond` for defining scripts:
+
+- listeners are registered with **regular expressions**, which can be cumbersome and error-prone, and encourage authors to design scripts with unpredictable interfaces.  
```

Comment:
Huge :+1: to this as a problem. I find myself wishing all the time that I could define commands as something more like optparse.